src/components/views/rooms/RoomList.js

```diff
@@ -94,10 +101,12 @@ module.exports = React.createClass({
                 }
                 this._groupStores[tag] = GroupStoreCache.getGroupStore(tag);
                 this._groupStores[tag].registerListener(() => {
-                    this.updateSelectedTagsEntities();
+                    // This group's rooms or members may have updated, update rooms for its tag
+                    this.updateSelectedTagsRooms(dmRoomMap, [tag]);
                 });
             });
-            this.updateSelectedTagsEntities();
+            // Filters themselves have changed, refresh the selected tags
```

Comment:
ok but the mapping from tag to room hasn't changed, so why do we need to rebuild `_selectedTagsRoomIdsForGroup` ?

Reply:
Oh, you're right - we should be passing `[]` there.

Reply:
Right, so I see what you were saying about having two methods.

Reply:
:grin: